lib/util.js

```diff
@@ -241,6 +242,41 @@ function inspectPromise(p) {
 
 
 function formatValue(ctx, value, recurseTimes) {
+
+  if (ctx.showProxy &&
+      (typeof value === 'object' || typeof value === 'function')) {
+    var proxy = undefined;
+    var proxyCache = ctx.proxyCache;
+    if (!proxyCache)
+      proxyCache = ctx.proxyCache = new WeakMap();
+    // Determine if we've already seen this object and have
+    // determined that it either is or is not a proxy.
+    if (proxyCache.has(value)) {
+      // We've seen it, if the value is not undefined, it's a Proxy.
+      proxy = proxyCache.get(value);
+    } else {
+      // Haven't seen it. Need to check.
+      // If it's not a Proxy, this will return undefined.
+      // Otherwise, it'll return an array. The first item
+      // is the target, the second item is the handler.
+      // We ignore (and do not return) the Proxy isRevoked property.
+      proxy = binding.getProxyDetails(value);
+      if (proxy) {
+        // We know for a fact that this isn't a Proxy.
+        // Mark it as having already been evaluated.
+        proxyCache.set(proxy, undefined);
```

Comment:
Why are you storing the array that .getProxyDetails() returns as a key?  I would expect this to be `proxyCache.set(value, proxy)`.  The comment seems wrong as well.  This whole `if` statement seems like it could be dropped, actually.